### PR TITLE
How to distinguish between no data and a timeout

### PR DESCRIPTION
**Pull request by awang**
cc @krishmanoh2

### CODE REVIEW

File: doc/user/content/sql/tail.md [hunk 1]
```diff
@@ -195,6 +195,9 @@ It is `true` if no more timestamps will appear that are strictly less than the
 timestamp.
 All further columns after `progressed` will be `NULL` in the `true` case.
 
+`mz_progressed` can be used to distinguish between a result with that contains
+no data (`true`) and a timeout (no rows returned).
```
Comment (benesch) [inline doc/user/content/sql/tail.md:199]: So this doesn't add anything that's not already covered here (though `progressed` is a typo now; I'll fix that in another PR). "a result" here is too imprecise to be useful, I think.

Certainly we should try to understand why this doc failed our users, but the next paragraph is already covers exactly this situation, IMO.

**Review by benesch — CHANGES_REQUESTED** [review]

File: doc/user/content/sql/tail.md (hunk 1 above)
Comment (awang) [reply doc/user/content/sql/tail.md:199]: Hmm, i see your point! I can close this PR and open a new one.  I wonder if it's a combination of a few things: 

1. we were all reading the doc really quickly, and were overly focused specifically on the "initial snapshot" scenario 
2. the `Without...it is impossible` phrasing caused my brain to skim over the section a bit

**Comment by awang** [comment]
Does this look better?  cc @msudberg 

(sorry, I know I said I'd open a new PR, but i decided that it's better to update this one to preserve the conversation history)

**Review by benesch — COMMENTED** [review]
So, when I wrote this text, I was intending to capture two things: this paragraph is meant to convey intuition rather than precise semantics, and that you actually _need_ these messages in order to be able to determine whether Materialize has stalled or has actually produced no updates. I'm down to rephrase this, but the current proposal presently drops those two bits, and I do think are important to retain in some form.

**Comment by benesch** [comment]
Maybe we can cleave the paragraph in half instead? One sentence like "Intuitively, progress messages ..." and another paragraph that really hits you over the head with why these messages are important.

**Comment by msudberg** [comment]
@awang - I think the current phrasing is fine, but it should be the opening paragraph of the `PROGRESS` section instead of inserted into the middle of the `mz_progressed` explanation.

**Comment by awang** [comment]
thanks all!  hopefully this does the trick?

if necessary, I like Nikhil's idea of adding some more text that reiterates things to try and make it super obvious.

**Review by petrosagg — APPROVED** [review]

**Comment by petrosagg** [comment]
@benesch I picked this up and fixed the lint error. Should we merge this?

**Comment by benesch** [comment]
Thanks! I also reworked to take @msudberg's suggestion: rather than cleaving the paragraph in two, I just moved it to the top of the section.

**Pull request merged**